### DISCUSSION TITLE
Checking Botan::Kyber against Known Answer Test

### DISCUSSION THREAD

**Discussion by yanboti** [randombit/botan#3857] [Q&A] [answered]
Dear Botan community,

I would like to be able to check Botan's implementation against the KAT that can be found [here](https://pq-crystals.org/kyber/resources.shtml).
However, I'm not sure how to remove the RNG requirement for the encapsulation function in  src/examples/kyber:L21.
Is there a way to remove the RNG from this function, and also from the decapsulation function?

Thank you for your help and happy holidays!

**Comment by randombit** [comment, marked as answer]
You would not necessarily *remove* the RNG but rather replace it with one that outputs exactly the bytes that the KAT expects. That is how we test Kyber against the KATs during the test suite: https://github.com/randombit/botan/blob/master/src/tests/test_kyber.cpp#L138-L147

**Reply by yanboti** [reply]
Thank you for the answer. This is very helpful!
